graphql-dgs-codegen-core/src/main/kotlin/com/netflix/graphql/dgs/codegen/generators/java/TypeUtils.kt

```diff
@@ -51,18 +46,33 @@ class TypeUtils(private val packageName: String, private val config: CodeGenConf
         "PresignedUrlResponse" to ClassName.get("com.netflix.graphql.types.core.resolvers", "PresignedUrlResponse"),
         "Header" to ClassName.get("com.netflix.graphql.types.core.resolvers", "PresignedUrlResponse", "Header"))
 
-    fun findReturnType(fieldType: Type<*>): JavaTypeName {
+    fun findReturnType(fieldType: Type<*>, useInterfaceType: Boolean = false, useWildcardType: Boolean = false): JavaTypeName {
         val visitor = object : NodeVisitorStub() {
             override fun visitTypeName(node: TypeName, context: TraverserContext<Node<Node<*>>>): TraversalControl {
-                val typeName = node.toJavaTypeName()
+                val typeName = node.toJavaTypeName(useInterfaceType)
                 val boxed = boxType(typeName)
                 context.setAccumulate(boxed)
                 return TraversalControl.CONTINUE
             }
             override fun visitListType(node: ListType, context: TraverserContext<Node<Node<*>>>): TraversalControl {
                 val typeName = context.getCurrentAccumulate<JavaTypeName>()
                 val boxed = boxType(typeName)
-                val parameterizedTypeName = ParameterizedTypeName.get(ClassName.get(List::class.java), boxed)
+
+                var canUseWildcardType = false
+                if (useWildcardType) {
```

Comment:
What does useWildcardType indicate?

Reply:
Hi @srinivasankavitha the `useWildcardType` is used to generate `? extends X` like in:

```
public interface IBookPage {
    List<? extends IBook> getItems();
    IPageInfo getPageInfo();
}
```